### ISSUE TITLE
Error messages too generic

### ISSUE THREAD

**Issue by irgijs** [markdirish/node-odbc#43]
Hi there,

When executing a query which results in a database error (e.g. if there is something wrong with the syntax of the query), the odbc package does throw an exception as expected. The exception does contain a very generic error message however.

Example:

```
const statement = await connection.createStatement();
await statement.prepare("select * from nonexistingtable");
await statement.execute();
``` 

I expect the abovementioned code to throw this error message:
`Invalid object name 'nonexistingtable'.`

Yet, node-odbc throws this error message 
`[odbc] Error executing the statement`

Is there any way to retrieve the error which caused the exception thrown by node-odbc?

**Comment by markdirish** [comment]
Hello!

There should be a property on the error object with the name 'odbcErrors'. This is an array that returns all of the results from calling`SQLGetDiagRec` and each entry in the array includes an error message, the SQL code, and the SQL state.

Even this solution isn't perfect, and I'm open to suggestions. The 1.x version would make the main error the first result from getting to ODBC errors and then include any error after that in an array similar to mine...

**Comment by irgijs** [comment]
Hi Mark,

Thanks for you quick response! Retrieving these errors from the 'odbcErrors' seems okay for now, but it is not set on the error object.

The error object only contains the `stack` and `message` properties.  Getting the `odbcErrors` results in undefined:

````
	try {
                // < preparing statement with incorrect SQL >
		const result = await statement.execute();
	} catch (error) {
		console.log(error.odbcErrors); // undefined
	}
```

**Comment by MrTeapot** [comment]
Hey!

I think the issue is in Statement.js.
```
  /**
   * Executes the prepared SQL statement template with the bound parameters, returning the result.
   * @param {function} [callback] - The callback function that returns the result. If omitted, uses a Promise.
   */
  execute(callback = undefined) {
    if (typeof callback !== 'function' && typeof callback !== 'undefined') {
      throw new TypeError('[node-odbc]: Incorrect function signature for call to statement.execute({function}[optional]).');
    }

    if (typeof callback === 'undefined') {
      return new Promise((resolve, reject) => {
        this.odbcStatement.execute((error, result) => {
          if (error) {
            reject(new Error(error));
          } else {
            resolve(result);
          }
        });
      });
    }
```
Should be:

```
  /**
   * Executes the prepared SQL statement template with the bound parameters, returning the result.
   * @param {function} [callback] - The callback function that returns the result. If omitted, uses a Promise.
   */
  execute(callback = undefined) {
    if (typeof callback !== 'function' && typeof callback !== 'undefined') {
      throw new TypeError('[node-odbc]: Incorrect function signature for call to statement.execute({function}[optional]).');
    }

    if (typeof callback === 'undefined') {
      return new Promise((resolve, reject) => {
        this.odbcStatement.execute((error, result) => {
          if (error) {
            reject(error);
          } else {
            resolve(result);
          }
        });
      });
    }
```

**Comment by MrTeapot** [comment]
@markdirish Hey, what do you think about this? Is it a fix?

**Comment by markdirish** [comment]
Yep, this was an error with Promises calling `new Error()`, which was deleting all of the ODBC error information. Fixed in [this PR](https://github.com/markdirish/node-odbc/pull/72), available on 2.3.0 beta branch (coming to a release branch soon!)
